changelog.d/6-federation/fs-1179

```diff
@@ -0,0 +1 @@
+Removing a federation domain will now remove all conversations for that domain.
```

Comment:
maybe add an entry in 5-internal that explains how background-worker had to be rewired to allow for proper cleanup of de-federated domains?  is there other rewiring happening in this PR?